x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/common/validation/SourceDestValidator.java

```diff
@@ -419,6 +426,46 @@ public void validate(Context context, ActionListener<Context> listener) {
         }
     }
 
+    public static class RemoteClusterMinimumVersionValidation implements SourceDestValidation {
+
+        private final Version minExpectedVersion;
+
+        public RemoteClusterMinimumVersionValidation(Version minExpectedVersion) {
+            this.minExpectedVersion = minExpectedVersion;
+        }
+
+        @Override
+        public void validate(Context context, ActionListener<Context> listener) {
+            Set<String> remoteClusterNames = context.getRemoteClusterService().getRegisteredRemoteClusterNames();
+            Map<String, Version> remoteClusterVersions;
+            try {
+                remoteClusterVersions =
+                    remoteClusterNames.stream()
+                        .collect(toMap(
+                            cluster -> cluster,
+                            cluster -> context.getRemoteClusterService().getConnection(cluster).getVersion()));
+            } catch (NoSuchRemoteClusterException e) {
+                context.addValidationError(e.getMessage());
+                listener.onResponse(context);
+                return;
+            } catch (Exception e) {
+                context.addValidationError(ERROR_REMOTE_CLUSTER_SEARCH, e.getMessage());
+                listener.onResponse(context);
+                return;
+            }
+            Optional<Map.Entry<String, Version>> minRemoteClusterVersion =
+                remoteClusterVersions.entrySet().stream().min(comparingByValue(Version::compareTo));
+            if (minRemoteClusterVersion.isPresent()) {
+                if (minRemoteClusterVersion.get().getValue().before(minExpectedVersion)) {
+                    context.addValidationError(
+                        REMOTE_CLUSTERS_TOO_OLD,
+                        minExpectedVersion, minRemoteClusterVersion.get().getKey(), minRemoteClusterVersion.get().getValue());
+                }
+            }
```

Comment:
I think it would not only be good to return a why we need a cluster version of at least X, but also which clusters are too old. It could be more than one.

Reply:
Done.